modules/apps/29-fee/types/msgs.go

```diff
@@ -59,6 +59,48 @@ func (msg MsgRegisterCounterpartyAddress) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{signer}
 }
 
+// NewMsgRegisterDistributionAddress creates a new instance of MsgRegisterDistributionAddress
+func NewMsgRegisterDistributionAddress(portID, channelID, address, distributionAddress string) *MsgRegisterDistributionAddress {
+	return &MsgRegisterDistributionAddress{
+		Address:             address,
+		DistributionAddress: distributionAddress,
+		PortId:              portID,
+		ChannelId:           channelID,
+	}
+}
+
+// ValidateBasic implements sdk.Msg and performs basic stateless validation
+func (msg MsgRegisterDistributionAddress) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return sdkerrors.Wrap(err, "failed to convert msg.Address into sdk.AccAddress")
+	}
```

Comment:
I need to add the same for `msg.DistributionAddress`

Reply:
its also worth checking here that `msg.Address != msg.DistributionAddress` since that is a waste of a message/mapping

Reply:
Done